miden-tx/src/tests/kernel_tests/test_account.rs

```diff
@@ -441,6 +449,100 @@ fn test_set_map_item() {
     );
 }
 
+#[test]
+fn test_storage_offset() {
+    // setup assembler
+    let assembler = TransactionKernel::assembler_testing();
+
+    // setup account
+    let id = AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_OFF_CHAIN).unwrap();
+    let vault = AssetVault::mock();
+    let storage = AccountStorage::new(vec![], BTreeMap::new()).unwrap();
+    let source_code = "
+        use.miden::account
+        use.kernel::memory
+
+        export.set_item
+            push.4.5.6.7.0
+            exec.account::set_item
+            dropw dropw
+        end
+
+        export.get_item_0
+            push.0
+            exec.account::get_item
+            swapw dropw swapw
+        end
+
+        export.get_item_1
+            push.1
+            exec.account::get_item
+            swapw dropw
+        end
+    ";
+    let code = AccountCode::mock_specific(source_code, assembler.clone());
+
+    // modify procedure offsets
+    let procedures_with_offsets = vec![
+        AccountProcedureInfo::new(*code.procedures()[0].mast_root(), 2),
+        AccountProcedureInfo::new(*code.procedures()[1].mast_root(), 1),
+        AccountProcedureInfo::new(*code.procedures()[2].mast_root(), 2),
+    ];
+
+    // rebuild [AccountCode] using new procedures with offsets
+    let code = AccountCode::from_parts(code.mast().clone(), procedures_with_offsets.clone());
```

Comment:
I would add a `TODO` somewhere here indicating that we are doing this only because we can't currently specify storage offsets via MASM code directly.